boards/wio_terminal/examples/wifi_connect.rs

```diff
@@ -0,0 +1,144 @@
+        &mut peripherals.MCLK,
+        &mut peripherals.OSC32KCTRL,
+        &mut peripherals.OSCCTRL,
+        &mut peripherals.NVMCTRL,
+    );
+    let mut delay = Delay::new(core.SYST, &mut clocks);
+    let mut sets: Sets = Pins::new(peripherals.PORT).split();
+
+    // Set up the display so we can print out APs.
+    let (mut display, _backlight) = sets
+        .display
+        .init(
+            &mut clocks,
+            peripherals.SERCOM7,
+            &mut peripherals.MCLK,
+            &mut sets.port,
+            24.mhz(),
+            &mut delay,
+        )
+        .unwrap();
+    clear(&mut display);
+    let mut textbuffer = String::<U256>::new();
+
+    let mut user_led = sets.user_led.into_open_drain_output(&mut sets.port);
+    user_led.set_low().unwrap();
+
+    // Initialize the wifi peripheral.
+    let args = (
+        sets.wifi,
+        peripherals.SERCOM0,
+        &mut clocks,
+        &mut peripherals.MCLK,
+        &mut sets.port,
+        &mut delay,
+    );
+    let nvic = &mut core.NVIC;
+    disable_interrupts(|cs| unsafe {
+        wifi_init(cs, args.0, args.1, args.2, args.3, args.4, args.5).unwrap();
+        WIFI.as_mut().map(|wifi| {
+            wifi.enable(cs, nvic);
+        });
+    });
+
+    let version = unsafe {
+        WIFI.as_mut()
+            .map(|wifi| wifi.blocking_rpc(rpc::GetVersion {}).unwrap())
+            .unwrap()
+    };
+    writeln!(textbuffer, "fw: {}", version).unwrap();
+    write(
+        &mut display,
+        textbuffer.as_str(),
+        Point::new(320 - (3 + version.len() * 12) as i32, 3),
+    );
+    textbuffer.truncate(0);
+
+    let mac = unsafe {
+        WIFI.as_mut()
+            .map(|wifi| wifi.blocking_rpc(rpc::GetMacAddress {}).unwrap())
+            .unwrap()
+    };
+    writeln!(textbuffer, "mac: {}", mac).unwrap();
+    write(&mut display, textbuffer.as_str(), Point::new(3, 3));
+    textbuffer.truncate(0);
+
+    let ip_info = unsafe {
+        WIFI.as_mut()
+            .map(|wifi| {
+                wifi.connect_to_ap(
+                    &mut delay,
+                    "NETWORK_NAME",
+                    "PASSWORD_HERE",
+                    Security::WPA2_SECURITY | Security::AES_ENABLED,
+                )
+                .unwrap()
+            })
+            .unwrap()
+    };
+    user_led.set_high().ok();
+    writeln!(textbuffer, "ip = {}", ip_info.ip).unwrap();
+    write(&mut display, textbuffer.as_str(), Point::new(3, 30));
+    textbuffer.truncate(0);
+    writeln!(textbuffer, "netmask = {}", ip_info.netmask).unwrap();
+    write(&mut display, textbuffer.as_str(), Point::new(3, 42));
+    textbuffer.truncate(0);
+    writeln!(textbuffer, "gateway = {}", ip_info.gateway).unwrap();
+    write(&mut display, textbuffer.as_str(), Point::new(3, 54));
+    textbuffer.truncate(0);
+
+    loop {
+        user_led.toggle();
+        delay.delay_ms(200u8);
+    }
+}
+
+wifi_singleton!(WIFI);
+
+fn clear(display: &mut wio::LCD) {
```

Comment:
Newer versions of embedded-graphics have a clear function which may or may not be more efficient.

Reply:
https://docs.rs/embedded-graphics/0.6.2/embedded_graphics/trait.DrawTarget.html#method.clear

Reply:
Done.